### DISCUSSION TITLE
Performance Planetscale compare to DO mysql

### DISCUSSION THREAD

**Discussion by almukhalafi** [planetscale/discussion#492] [Q&A]
Hi PlanetScale community,

I recently migrated my database from DigitalOcean MySQL Managed Database to PlanetScale. The migration went smoothly, but I'm seeing some performance problems. When I run a query to select 50K rows, PlanetScale is significantly slower than DigitalOcean.

I've checked the query plan and it looks like the query is being optimized correctly. I've also checked the distribution of the data and it's evenly distributed across the nodes. I'm running PlanetScale in a region that is close to my users.

I'm not sure what else to check. Can anyone help me to understand why PlanetScale is slower than DigitalOcean for this particular workload?
 
Thanks,

**Comment by orware** [comment]
Hello @almukhalafi,

I believe you and I worked on a recent ticket together where you were moving a database away from PlanetScale?

I'm not sure which database you may be referencing here, but I seem to only see one Hobby plan database associated with the account I worked with on that earlier ticket so that alone could provide an explanation, as the Hobby plan databases are only assigned limited resources.

Aside from that reason alone being a potential explanation, querying for a result set containing 50K rows is going to be somewhat heavy regardless.

The other factor could potentially be related to physical distance between your location and the original database and your location and the PlanetScale database. If the distance between your location and the PlanetScale database is further than before, that would add extra latency as well, especially when querying for a larger result set.

I hope the above is helpful information but if more assistance is needed please submit a ticket via https://support.planetscale.com/ so that we can continue working with you in more detail.

**Reply by re-thc** [reply]
Is there any documentations on quantifying what limited resources mean? Does that differ for every plan? How many resources are assigned? Are there benchmarks on these?

**Reply by orware** [reply]
Not at this time, but our paid plans do receive additional resources.

We do have some synthetic benchmarks demonstrating what is possible within our platform, such as the [One Million Connections](https://planetscale.com/blog/one-million-connections) blog post, but that type of post is more useful for larger Enterprise type customers to see in most cases.

For smaller scale customers, the experience can be mixed because the workloads and schemas vary between users so that can impact things significantly. 

As an example of that, we've worked with users coming in where the database is behaving poorly in their eyes, but after some review of the data in [Insights](https://planetscale.com/docs/concepts/query-insights) of the queries being run, and the schema, some suggestions for indexing were uncovered and after adding those performance improved dramatically. So a [well-optimized schema](https://planetscale.com/courses/mysql-for-developers) can go a long way to help avoid performance issues.
